datalad/cmd.py

```diff
@@ -674,41 +683,3 @@ def run(self, cmd, env=None, *args, **kwargs):
         # All communication here will be returned as unicode
         # TODO: do that instead within the super's run!
         return assure_unicode(out), assure_unicode(err)
-
-
-# ####
-# Preserve from previous version
-# TODO: document intention
-# ####
-# this one might get under Runner for better output/control
-def link_file_load(src, dst, dry_run=False):
-    """Just a little helper to hardlink files's load
-    """
-    dst_dir = op.dirname(dst)
-    if not op.exists(dst_dir):
-        os.makedirs(dst_dir)
-    if op.lexists(dst):
-        lgr.log(9, "Destination file %(dst)s exists. Removing it first", locals())
-        # TODO: how would it interact with git/git-annex
-        unlink(dst)
-    lgr.log(9, "Hardlinking %(src)s under %(dst)s", locals())
-    src_realpath = op.realpath(src)
-
-    try:
-        os.link(src_realpath, dst)
-    except AttributeError as e:
-        lgr.warn("Linking of %s failed (%s), copying file" % (src, e))
-        shutil.copyfile(src_realpath, dst)
-        shutil.copystat(src_realpath, dst)
-    else:
-        lgr.log(2, "Hardlinking finished")
-
-
-def get_runner(*args, **kwargs):
```

Comment:
FWIW, pruning/moving API which is used by extensions without lengthy deprecation cycle brings pain to the developers and users. Primary reason - absent ability to declare Breaks for pypi and conda. So now, when we release datalad, we must release crawler, but even then who had older crawler extension installed, and updates only datalad would be screwed... Could be even at the level of just incoming datalad commands. I truly "enjoyed" similar situation with keyring and keyrings.alt doing similar act.